generator/osm_source.hpp

```diff
@@ -48,6 +51,17 @@ class EmitterBase
   virtual void GetNames(vector<string> & names) const = 0;
 };
 
+class SyncOfstream
```

Comment:
SyncOfstream  is not a nested class. Do you suggest to move it a separate file? At first let's decide if we need SyncOfstream at all. Have a look please as Polygonizer. There's a method in it:
```c++
    void operator () (FeatureBuilder1 & fb, SyncOfstream & featureId2osmIds)
    {
      buffer_vector<borders::CountryPolygons const *, 32> vec;
      m_countries.ForEachInRect(fb.GetLimitRect(), InsertCountriesPtr(vec));

      switch (vec.size())
      {
      case 0:
        break;
      case 1:
        EmitFeature(vec[0], fb, featureId2osmIds);
        break;
      default:
        {
#if PARALLEL_POLYGONIZER
          m_ThreadPoolSemaphore.acquire();
          m_ThreadPool.start(new PolygonizerTask(this, vec, fb, featureId2osmIds));
#else
          PolygonizerTask task(this, vec, fb, featureId2osmIds);
          task.RunBase();
#endif
        }
      }
    }
```
As far as I see in case 2 PolygonizerTask(s) could be executed in a separate thread. So EmitFeature could be called from different threads. To be on the save side I decided to use mutex (SyncOfstream). Do you think EmitFeature() couldn't be called from different threads?